blog/2022-12-07-build-a-frontend-dapp-for-celo-network-in-angular/2022-12-07-build-a-frontend-dapp-for-celo-network-in-angular.md

```diff
@@ -0,0 +1,1698 @@
+
+Copy the 00-deploy.js file and paste it to a new file called 01-deploy-NFTAuction.js. 
+
+Your hardhat-deploy plugin deploys your contracts serially using the naming of the file. So, when you run the deploy command, it will run the 00-deploy file first, then run the 01-deploy-nftauction.js file next.
+
+Now open the 01-deploy-NFTAuction.js file.
+
+Update the code to deploy the NFTAuctionManager Contract.
+
+Your code should look like this
+
+```js
+
+await deploy("NFTAuctionManager", {
+from: deployer,
+args: [],
+log: true,
+})
+
+module.exports.tags = ["NFTAuctionManager"];
+
+```
+
+
+
+
+Deploy the Contracts by running the following commands on any terminal (make sure you are in the packages/hardhat directory)
+
+```text
+npx hardhat deploy –network alfajores
+```
+
+
+
+If all is well, you should see a message from hardhat with the transaction hash of your Contract deployment and the address of your new Contract
+
+You can now view your contract on the CELO explorer (Alfajores) using the address.
+
+Now that we have deployed our contract let's build the dApp.
+
+Before we go on, Our dApp will be running in the browser, so you will need a wallet to test within the browser. If you have not yet installed Metamask, this will be a good point to do that.
+
+Also, once you have Metamask installed, ensure you add the CELO Networks - Mainnet and Alfajores ( TestNet ) - to it. If you are unsure how to go about this, You can use this tutorial ( <https://developers.celo.org/3-simple-steps-to-connect-your-metamask-wallet-to-celo-732d4a139587> ) as a guide.
+
+
+
+# DAPP Frontend
+
+Switch to the angular project in your terminal
+
+```text
+cd packages/angular-app
+```
+
+
+Now install the needed packages
+
+```text
+npm i
+```
+
+
+Once the packages install, let's run the code to see how it looks first with the base Celo composer code
+
+![image](images/2.png)
+
+Click on the connect button, and you will see how it works for now.	
+
+You can now connect to a crypto wallet, and the response from your connection is also displayed back to you. Now, let's go through the code that produced that.
+
+Open **src/app/core/web3.ts**
+
+The first thing we need to do is figure out a way to get access to our Web3 provider.
+
+This Web3 provider allows your application to communicate with an Ethereum Node. Providers take JSON-RPC requests and return the response. 
+
+
+
+The /core/web3 file contains an injection token that allows you to get access to the web provider.	
+
+
+```js
+
+import { InjectionToken } from '@angular/core';
+import Web3 from 'web3';
+
+export const WEB3 = new InjectionToken<Web3>('web3', {
+  providedIn: 'root',
+  factory: () => new Web3(Web3.givenProvider)
+});
+
+```
+
+Now that we know how to get access to the web3 provider, Most of your interactions with the blockchain are going to be happening through the web3Service so let's go through that also.
+
+Open the **packages\angular-app\src\app\services\contract\web3.services.ts** file 
```

Comment:
Open the `**packages\angular-app\src\app\services\contract\web3.services.ts**` file